File: connection-examples/rust/src/subscribe.rs
use crate::connection::create_client;

/// Run a subscribe over the PUBNUB materialized view
pub(crate) fn subscribe() {
    let mut client = create_client().expect("Error creating client.");
    let mut transaction = client.transaction().expect("Error creating transaction.");
    transaction.execute("DECLARE c CURSOR FOR SUBSCRIBE (SELECT symbol, bid_price::text FROM market_orders) WITH (SNAPSHOT = false);", &[]).expect("Error creating cursor.");

    loop {
        let results = transaction.query("FETCH ALL c;", &[]).expect("Error running fetch.");
        for row in results {
            println!("{:} - {:}", row.get::<usize, String>(2), row.get::<usize, String>(3));
        }
    }
}

File: connection-examples/php/subscribe.php
<?php
// Include the Postgres connection details
require 'connection.php';
require 'state.php';

// Begin a transaction
$connection->beginTransaction();
// Declare a cursor
$statement = $connection->prepare('DECLARE c CURSOR FOR SUBSCRIBE (SELECT sum FROM counter_sum) WITH (PROGRESS);');
// Execute the statement
$statement->execute();

// Create a new State object
$state = new State(true);

// Buffer updates
$buffer = array();

/* Fetch all of the remaining rows in the result set */
while (true) {
    $subscribe = $connection->prepare('FETCH ALL c');
    $subscribe->execute();
    $result = $subscribe->fetchAll(PDO::FETCH_ASSOC);

    // Iterate over the results using a foreach loop
    foreach ($result as $row) {
        // Access the values of each column in the row using the column name
        $ts = $row['mz_timestamp'];
        $progressed = $row['mz_progressed'];
        $diff = $row['mz_diff'];
        $sum = $row['sum'];

        if ($progressed == 1) {
            $state->update($buffer, $ts);
            $buffer = array();
            print_r($state->get_state());
        } else {
            array_push($buffer, new Update(
                array('sum' => $sum),
                $diff
            ));
        }
    }
}